spec/policy_machine_storage_adapters/active_record_spec.rb

```diff
@@ -1523,6 +1523,70 @@
     end
   end
 
+  describe '#associations_filtered_by_operation' do
+    let!(:new_pm) { PolicyMachine.new(name: 'AR PM', storage_adapter: PolicyMachineStorageAdapter::ActiveRecord) }
+
+    let(:operation_1) { new_pm.create_operation('operation_1') }
+    let(:operation_2) { new_pm.create_operation('operation_2') }
+    let(:operation_3) { new_pm.create_operation('operation_3') }
+    let(:opset_1) { new_pm.create_operation_set('operation_set_1') }
+    let(:opset_2) { new_pm.create_operation_set('operation_set_2') }
+    let(:opset_3) { new_pm.create_operation_set('operation_set_3') }
+
+    let(:user_attr) { new_pm.create_user_attribute('user_attr') }
+
+    let(:object_attr_1) { new_pm.create_object_attribute('object_attr_1') }
+    let(:object_attr_2) { new_pm.create_object_attribute('object_attr_2') }
+    let(:object_attr_3) { new_pm.create_object_attribute('object_attr_3') }
+
+    before do
+      new_pm.add_assignment(opset_1, operation_1)
+      new_pm.add_assignment(opset_2, operation_2)
+      new_pm.add_assignment(opset_3, operation_3)
+
+      new_pm.add_association(user_attr, opset_1, object_attr_1)
+      new_pm.add_association(user_attr, opset_2, object_attr_2)
+      new_pm.add_association(user_attr, opset_3, object_attr_3)
+
+      new_pm.add_assignment(opset_1, opset_2)
+    end
+
+    let(:peas) { PolicyMachineStorageAdapter::ActiveRecord::PolicyElementAssociation.all }
+    let(:pea_1) { peas.find_by(operation_set_id: opset_1.id) }
+    let(:pea_2) { peas.find_by(operation_set_id: opset_2.id) }
+    let(:pea_3) { peas.find_by(operation_set_id: opset_3.id) }
+
+    context 'when provided no policy element associations' do
```

Comment:
What's with the "when"s in these context block descriptions? I can't remember seeing you do that before.

Reply:
Wes hit me with the http://www.betterspecs.org/